python/lsst/ts/watcher/rules/electric_generator_fail.py

```diff
@@ -0,0 +1,181 @@
+type: object
+properties:
+    name_master:
+        description: >-
+            CSC name and index in the form `name:index`
+            for the master electric generator.
+        type: string
+    name_slave:
+        description: >-
+            CSC name and index in the form `name:index`
+            for the slave electric generator.
+        type: string
+    severity_individual_fail:
+        description: >-
+            Alarm severity for when a electric generator has
+            a main failure.
+        type: string
+        default: {AlarmSeverity.SERIOUS.name}
+        enum:
+{severity_values}
+    severity_both_fail:
+        description: >-
+            Alarm severity for when both electric generators
+            have a main failure.
+        type: string
+        default: {AlarmSeverity.CRITICAL.name}
+        enum:
+{severity_values}
+required:
+  - name_master
+  - name_slave
+additionalProperties: false
+       """
+        return yaml.safe_load(schema_yaml)
+
+    def compute_alarm_severity(
+        self, data: salobj.BaseMsgType, **kwargs: typing.Any
+    ) -> AlarmSeverityReasonType:
+        """Compute and set alarm severity and reason.
+
+        Parameters
+        ----------
+        data : `salobj.BaseMsgType`
+              Message from the topic described by topic_callback.
+        **kwargs : `dict` [`str`, `typing.Any`]
+            Keyword arguments. If triggered by `TopicCallback` calling
+            `update_alarm_severity`, the arguments will be as follows:
+
+            * topic_callback : `TopicCallback`
+              Topic callback wrapper.
+
+        Returns
+        -------
+        None, if no change or unknown, or a tuple of two values:
+
+        severity: `lsst.ts.xml.enums.Watcher.AlarmSeverity`
+            The new alarm severity.
+        reason : `str`
+            Detailed reason for the severity, e.g. a string describing
+            what value is out of range, and what the range is.
+            If ``severity`` is ``NONE`` then this value is ignored (but still
+            required) and the old reason is retained until the alarm is reset
+            to ``nominal`` state.
+
+        Notes
+        -----
+        You may return `NoneNoReason` if the alarm state is ``NONE``.
+        """
+        severity, reason = NoneNoReason
+
+        genset_master_failure = getattr(data, "mainFailure")
+
+        if genset_master_failure is True:
+            if self._remote_index_master == data.salIndex:
+                self._master_main_failure = True
+                remote_name = self._remote_name_master
+                remote_index = self._remote_index_master
+            elif self._remote_index_slave == data.salIndex:
+                self._slave_main_failure = True
+                remote_name = self._remote_name_slave
+                remote_index = self._remote_index_slave
+
+            reason = f"{remote_name}:{remote_index} electric generator main failure."
```

Comment:
`electric generator main failure` -> `electric generator in main failure` to have a similar message as when both are in main failure.